keras_nlp/models/roberta/roberta_preprocessing.py

```diff
@@ -12,10 +12,174 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""RoBERTa preprocessing layers."""
+
 import tensorflow as tf
 import tensorflow_text as tf_text
 from tensorflow import keras
 
+from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class RobertaPreprocessor(keras.layers.Layer):
+    """RoBERTa preprocessing layer.
+
+    This preprocessing layer will do three things:
+
+    - Tokenize any number of inputs using a
+      `keras_nlp.tokenizers.BytePairTokenizer`.
+    - Pack the inputs together with the appropriate `"<s>"`, `"</s>"` and
+      `"<pad>"` tokens, i.e., adding a single `"<s>"` at the start of the
+      entire sequence, `"</s></s>"` at the end of each segment, save the last
+      and a `"</s>"` at the end of the entire sequence.
+    - Construct a dictionary with keys `"token_ids"` and `"padding_mask"` that
+      can be passed directly to a RoBERTa model.
+
+    This layer will accept either a tuple of (possibly batched) inputs, or a
+    single input tensor. If a single tensor is passed, it will be packed
+    equivalently to a tuple with a single element.
+
+    The `BytePairTokenizer` can be accessed via the `tokenizer` property on
+    this layer, and can be used directly for custom packing on inputs.
+
+    Args:
+        vocabulary: string or dict, maps token to integer ids. If it is a
+            string, it should be the file path to a json file.
+        merges: string or list, contains the merge rule. If it is a string,
```

Comment:
Is there a standard format for the merge rule file? I would have no idea how to construct this as a user.

Reply:
Do you think giving a reference here would be helpful?

Reply:
I think constructing the `merges` and `vocabulary` is basically fully unsupported at the library level right now. Right now, our answer is just to copy these files. We should open an issue to track this.

Abheesht already linked this issue a while back, has a little more color -> https://github.com/huggingface/transformers/issues/1083

Reply:
Cool! Will remove the reference to https://storage.googleapis.com/keras-nlp/models/roberta_base/merges.txt, then